runner/local/simple.go

```diff
@@ -85,52 +99,64 @@ func (r *simpleRunner) Erase(run runner.RunId) {
 	}
 }
 
-func (r *simpleRunner) abortImpl(run runner.RunId, state runner.ProcessState) runner.ProcessStatus {
+func (r *simpleRunner) updateStatus(new runner.ProcessStatus) runner.ProcessStatus {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	result, ok := r.running[run]
+	old, ok := r.runs[new.RunId]
 	if !ok {
-		return runner.BadRequestStatus(run, fmt.Errorf("could not find abortable run: %v", run))
+		return runner.BadRequestStatus(new.RunId, fmt.Errorf("cannot find run %v", new.RunId))
+	}
+	if old.State.IsDone() {
+		return old
 	}
-	result.Abort()
-	delete(r.running, run)
-	r.runs[run] = runner.AbortStatus(run)
-	return r.runs[run]
+	r.runs[new.RunId] = new
+	if new.State.IsDone() && r.running.runId == new.RunId {
```

Comment:
This section is confusing. If we only allow one run at a time, when would the old status indicate not done, the new indicate done, and have the runIds mismatched? And it seems weird to close the cancelCh unless we expect something to be listening, i.e. not after COMPLETE.